### PR TITLE
Use go driver bson lib

### PR DESCRIPTION
**Pull request by TomerYakir**

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -177,35 +237,44 @@ func TestServer(t *testing.T) {
...
+	client, err := mongo.NewClient(opts)
 	if err != nil {
-		t.Errorf("cannot dial: %s", err)
-		return
+		t.Errorf("cannot create a mongo client. err: %v", err)
 	}
-	defer session.Close()
 
-	coll := session.DB("test").C("bar")
+	ctx := context.Background()
```
Comment (tolsen) [inline server_test.go]: maybe we should have a timeout for the test?

Comment (TomerYakir) [reply server_test.go]: added!

File: server_test.go [hunk 2]
```diff
@@ -427,13 +428,15 @@ func TestServerWorkerWithContext(t *testing.T) {
 			t.Errorf("cannot create a mongo client. err: %v", err)
 			return
 		}
-
-		ctx := context.Background()
+		
+		ctx, cancelFunc := context.WithTimeout(context.Background(), 5 * time.Second)
```
Comment (tolsen) [inline server_test.go]: Is this overwriting the prior cancelFunc?  If so, will the previous defer run properly?  When is the name looked up?

Comment (TomerYakir) [reply server_test.go]: changed

**Pull request merged**